fbssdc/bpy.py

```diff
@@ -89,6 +125,23 @@ def main():
 
   subs = parser.add_subparsers(title='subcommands')
 
+
+  encode_full_parser = subs.add_parser('encode-dir', help='Encode a full directory (from .js).',
+                                  description='''Caveats:
+(1) dictionary misses are not supported yet
+(2) not all file sections are compressed natively yet;
+    the output should be compressed with Brotli
+''')
+  encode_full_parser.add_argument('binjs_encode',
+                             help='path to the binjs_encode binary')
+  encode_full_parser.add_argument('dictionary', type=argparse.FileType('rb'),
```

Comment:
what do you think of making dictionary arg optional so that it can default to an empty external dictionary

Reply:
Maybe in a followup bug. For the moment, I don't think that `format.write` supports dictionary as an optional argument.